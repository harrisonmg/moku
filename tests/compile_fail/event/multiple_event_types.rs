use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    use machine::BlinkyState;

    struct EventA;

    impl StateMachineEvent for EventA {}

    struct EventB;

    impl StateMachineEvent for EventB {}

    struct Top;

    impl TopState<BlinkyState, EventA> for Top {}
}

fn main() {}
