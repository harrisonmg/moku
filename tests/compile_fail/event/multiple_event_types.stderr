error: multiple `moku::StateMachineEvent`s are defined within this module
  --> tests/compile_fail/event/multiple_event_types.rs:18:5
   |
18 |     impl StateMachineEvent for EventB {}
   |     ^^^^
